src/components/FilterEditor.vue

```diff
@@ -67,7 +69,7 @@ export default class FilterEditor extends Vue {
   errors!: ErrorObject[];
 
   get conditionsTree() {
-    return buildConditionsEditorTree(this.filterTree);
+    return buildConditionsEditorTree(castFilterStepTreeValue(this.filterTree, this.columnTypes));
   }
```

Comment:
Now that this is done in the FilterEditor level, it should be removed from FilterStepForm.vue as well

Reply:
Btw, I think it's also used in the IFThenElseStepForm

Reply:
https://github.com/ToucanToco/weaverbird/pull/767/commits/2446543ebee2dc7a38204b3266a406b81e148104 https://github.com/ToucanToco/weaverbird/pull/767/commits/7dab6fbec53680c897c0804bc63be9dd075b73d6 https://github.com/ToucanToco/weaverbird/pull/767/commits/5789eae618450027d37ab79c52e9542908390277 https://github.com/ToucanToco/weaverbird/pull/767/commits/4e0a4e1d152291cfeb844a4800c47f5e44a656c5